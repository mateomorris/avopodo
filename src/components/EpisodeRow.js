import React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity, Alert } from 'react-native';

const styles = StyleSheet.create({
    container: {
        margin: 5, 
        shadowColor: '#000000',
        shadowOffset: {
            width: 0,
            height: 0
        },
        shadowRadius: 2,
        shadowOpacity: 0.75
    },
    thumbnail: { 
        width: 120, 
        height: 120, 
        borderRadius: 4
    }
});

export default class EpisodeRow extends React.Component {

    _getDate = (ms) => {
        const theDate = new Date(ms)
        const currentDate = new Date();

        const month = theDate.toLocaleString('en-us', { month: "long" })
        const day = theDate.getDate()
        const year = theDate.getFullYear();

        // Only show the year if it's past
        if (year == currentDate.getFullYear()) {
            return `${month} ${day}`;
        } else {
            return `${month} ${day}, ${year}`;
        }
    }

    render() {

        const { info, finished } = this.props
        const theDate = this._getDate(info.publishDate);

        return (
            <View opacity={finished ? 0.5 : 1}>
                <TouchableOpacity 
                style={{
                    borderBottomWidth: 1,
                    borderBottomColor: '#CCC',
                    padding: 20,
                    marginLeft: 10, 
                    marginRight: 10,
                    flexDirection: 'row',
                    height: 75
                }} onPress={this.props.onDetailPress}>
                    <View style={{
                        paddingRight: 10
                    }}>
                        <Text style={{
                            fontSize: 14,
                            fontWeight: "500",
                            color: '#666666'
                        }}>{theDate}</Text>
                        <Text style={{
                            fontSize: 16,
                            fontWeight: "600"
                        }} ellipsizeMode='tail' numberOfLines={1}>{info.title}</Text>
                    </View>
                    <TouchableOpacity style={{
                        paddingLeft: 20,
                        position: 'absolute',
                        right: 5,
                        alignItems: 'center',
                        alignContent: 'center',
                        justifyContent: 'center',
                        bottom: 0,
                        top: 0,
                        zIndex: 9
                    }} onPress={() => {
                        this.props.onPlayPress()
                    }}>
                        <Image 
                            source={require('../assets/play-gray.png')}
                            style={{
                                width: 20,
                                height: 20
                            }}
                        />
                    </TouchableOpacity>
                </TouchableOpacity>
            </View>
        );

    }
}