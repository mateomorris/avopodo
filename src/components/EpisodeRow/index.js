import React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity, Alert, ImageBackground } from 'react-native';
import * as Progress from 'react-native-progress';
import SvgUri from 'react-native-svg-uri';

import { CircleButton } from 'components/Button';
import icons from 'assets/generalIcons';

const styles = StyleSheet.create({
    container: {
        margin: 5, 
        shadowColor: '#000000',
        shadowOffset: {
            width: 0,
            height: 0
        },
        shadowRadius: 2,
        shadowOpacity: 0.75
    },
    thumbnail: { 
        width: 120, 
        height: 120, 
        borderRadius: 4
    }
});

export default class EpisodeRow extends React.Component {

    _getDate = (ms) => {
        const theDate = new Date(ms)
        const currentDate = new Date();

        const month = theDate.toLocaleString('en-us', { month: "long" })
        const day = theDate.getDate()
        const year = theDate.getFullYear();

        // Only show the year if it's past
        if (year == currentDate.getFullYear()) {
            return `${month} ${day}`;
        } else {
            return `${month} ${day}, ${year}`;
        }
    }

    render() {

        const { info, finished, buttonColor, playing } = this.props
        const theDate = this._getDate(info.publishDate);

        return (
            <View opacity={finished ? 0.5 : 1}>
                <TouchableOpacity 
                style={{
                    borderBottomWidth: 1,
                    borderBottomColor: '#CCC',
                    marginLeft: 10,
                    padding: 20,
                    paddingLeft: 10,
                    paddingRight: 10,
                    // marginLeft: 10, 
                    flexDirection: 'row',
                    justifyContent: 'space-between',
                    alignItems: 'center',
                    width: '100%',
                    height: 120
                }} onPress={this.props.onDetailPress}>
                    <View style={{
                        paddingRight: 45
                    }}>
                        <Text style={{
                            fontSize: 14,
                            fontWeight: "500",
                            color: '#666666'
                        }}>{theDate}</Text>
                        <Text style={{
                            fontSize: 16,
                            fontWeight: "600",
                            color: '#333'
                        }} ellipsizeMode='tail' numberOfLines={1}>{info.title}</Text>
                        <Text style={{
                            fontSize: 14,
                            fontWeight: "500",
                            color: '#666666'
                        }} ellipsizeMode='tail' numberOfLines={3}>{info.description}</Text>
                    </View>
                    <View style={{
                        paddingLeft: 10,
                        paddingRight: 25,
                        position: 'absolute',
                        right: 0,
                        alignItems: 'center',
                        alignContent: 'center',
                        justifyContent: 'center',
                        bottom: 0,
                        top: 0,
                        zIndex: 9,
                    }}>
                        <CircleButton 
                            onPress={() => {
                                this.props.onPlayPress()
                            }}
                            icon={playing ? icons.pause : icons.play}
                            size={20}
                            spinner={false}
                            color={buttonColor}
                        />
                    </View>
                </TouchableOpacity>
            </View>
        );

    }
}