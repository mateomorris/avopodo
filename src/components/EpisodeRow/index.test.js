import React from 'react';
import { Text } from 'react-native';
import EpisodeRow from 'components/EpisodeRow';
import renderer from 'react-test-renderer';

jest.useFakeTimers();

it('renders EpisodeRow correctly', () => {
  const tree = renderer
    .create(
      <EpisodeRow
        info={{
          audio: "https://play.podtrac.com/npr-381444908/npr.mc.tritondigital.com/NPR_381444908/media/anon.npr-mp3/npr/fa/2018/11/20181112_fa_fawpodmon.mp3?orgId=1&d=2894&p=381444908&story=667026316&t=podcast&e=667026316&ft=pod&f=381444908",
          description: `Every surface, every bit of air, every bit of water in your home is alive," says scientist Rob Dunn. His new book, 'Never Home Alone,' examines the bacteria, fungi, viruses, parasites and insects we live with â€” from armpit bacteria to black mold in our walls. Book critic Maureen Corrigan reviews 'A Ladder to the Sky' by John Boyne. She calls it "maliciously witty, erudite and ingeniously constructed."`,
          duration: 2894,
          id: "7fc9602c63904d4ab7c52517e2597b2b",
          publishDate: 1542058248000,
          showColor: "#030404",
          showDescription: "Fresh Air from WHYY, the Peabody Award-winning weekday magazine of contemporary arts and issues, is one of public radio's most popular programs. Hosted by Terry Gross, the show features intimate conversations with today's biggest luminaries.",
          showId: "1df990c9c80a4a04ba11d744c6c0f199",
          showImage: "https://d3sv2eduhewoas.cloudfront.net/channel/image/26a169d975804765aad223e90463a259.jpg",
          showPublisher: "NPR",
          showTitle: "Fresh Air",
          showWebsite: "http://www.npr.org/programs/fresh-air/?utm_source=listennotes.com&utm_campaign=Listen+Notes&utm_medium=website",
          title: "With Bugs & Bacteria Living In Your Home, You're 'Never Home Alone'"
        }}
        finished={true}
        testing={true}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});