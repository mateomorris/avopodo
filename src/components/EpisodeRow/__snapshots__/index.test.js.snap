// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EpisodeRow correctly 1`] = `
<View
  opacity={0.5}
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomColor": "#CCC",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 75,
        "justifyContent": "space-between",
        "marginLeft": 10,
        "opacity": 1,
        "padding": 20,
        "paddingRight": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "paddingRight": 35,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#666666",
            "fontSize": 14,
            "fontWeight": "500",
          }
        }
      >
        November 12
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#222",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        With Bugs & Bacteria Living In Your Home, You're 'Never Home Alone'
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "opacity": 1,
          "paddingLeft": 10,
          "paddingRight": 30,
          "position": "absolute",
          "right": 15,
          "top": 0,
          "zIndex": 9,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
            },
            Object {
              "opacity": 0.5,
              "position": "absolute",
            },
          ]
        }
      >
        <ARTSurfaceView
          style={
            Array [
              Object {
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <ARTShape
            d={
              Array [
                0,
                12,
                1.5,
                2,
                12,
                1.5,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0.011764705882352941,
                0.01568627450980392,
                0.01568627450980392,
                1,
              ]
            }
            strokeCap={0}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={3}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
        </ARTSurfaceView>
      </View>
    </View>
  </View>
</View>
`;
